docs/pages/dsVideoDevice/ds-video-device_L3_TestProcedure.md

```diff
@@ -0,0 +1,368 @@
+
+
+## Test Cases
+
+### dsVideoDevice_test1_FrameratePreChangePostChangeCallback_Verify.py
+
+#### Platform Supported - test01
+
+- Sink
+
+#### User Input Required - test01
+
+- No
+
+#### Acceptance Criteria - test01
+
+- Verify the display framerate change with pre and post change callbacks.
+
+#### Expected Results - test01
+
+- This test checks whether the callback is triggered before and after when the display framerate changes.
+
+#### Test Steps - test01
+
+- Run the test file: **`dsVideoDevice_test1_FrameratePreChangePostChangeCallback_Verify.py`**
+- Execution process:
+
+  Upon execution, the test will:
+
+  - Download all the required artifacts.
+  - Copy them to the target directory.
+  - Automatically start the test execution.
+
+- Iterating through supported display framerates:
+
+  The test will repeat for different supported display framerates. Before that set the FRF mode to `Disable`. And, then on setting each framerates callback need to be triggered accordingly.
+
+- Completion and results:
+
+  After all iterations are complete and based on the callback responses, the test will conclude and display the final result (PASS/FAIL).
+
+### dsVideoDevice_test2_SetAndGetZoomMode.py
+
+#### Platform Supported - test02
+
+- Source
+
+#### User Input Required - test0
+
+Yes (This will be automated later).
+
+#### Acceptance Criteria - test02
+
+Verify the changes in zoom modes across the supported zoom modes for designated video streams.
+
+#### Expected Results - test02
+
+The test will evaluate the various zoom modes. The user should notice a zoom mode change in video playback as different zoom modes are applied.
+
+#### Test Steps - test02
+
+- Select the test file:
+
+  - Run the Python script **`dsVideoDevice_test2_SetAndGetZoomMode.py`**
+
+- Execution process:
+
+  The test will:
+
+  - Download all the required artifacts and video streams.
+  - Copy them to the target directory.
+  - Automatically start the test execution.
+
+- Play the video stream:
+
+  Test starts the video stream playback specified in the test setup configuration file.
+
+- Iterating through supported zoom modes:
+
+  The test will repeat for different supported zoom modes. On setting each zoom mode user need to verify it is applied in video stream.
+
+- User interaction for verification:
+
+  For each iteration:
+
+  - The test will ask: "Is the selected Zoom mode applied in the device's output? (Y/N):"
+
+- Completion and results:
+
+  Once all necessary user actions are completed, the test will evaluate the results and display whether the test Passed or Failed.
+
+### dsVideoDevice_test3_SetDisplayFramerate.py
+
+#### Platform Supported - test03
+
+- Sink
+
+#### User Input Required - test03
+
+Yes (This will be automated later).
+
+#### Acceptance Criteria - test03
+
+Verify the impact of the supported display framerates for **Stream #2**.
```

Comment:
This is not matching with the test script

Reply:
updated